hadoop-hdds/server-scm/src/main/resources/webapps/scm/scm.js

```diff
@@ -32,40 +32,56 @@
-                    $scope.nodeStatus = nodeStatusCopy.slice(0, $scope.RecordsToDisplay);
+                        value.map(({key, value}) => {
+                            if(key == "HTTP") {
+                                protocol = key;
+                                portNo = value;
+                            }
+                            if(key == "HTTPS"){
+                                protocol = key.toLowerCase() === URLScheme ? key : "HTTP";
+                                portNo = value;
+                            }
+                        });
+                    return {
+                        hostname: key,
+                        opstate: value && value.find((element) => element.key == "OPSTATE").value,
+                        comstate: value && value.find((element) => element.key == "COMSTATE").value,
+                        portno: portNo,
+                        portval: protocol
+                    }});
+                nodeStatusCopy = [...$scope.nodeStatus];
+                $scope.lastIndex = Math.ceil(nodeStatusCopy.length / $scope.RecordsToDisplay);
+                $scope.nodeStatus = nodeStatusCopy.slice(0, $scope.RecordsToDisplay);
                 });
             /*if option is 'All' display all records else display specified record on page*/
-            $scope.UpdateRecordsToShow = () => {
-                 if($scope.RecordsToDisplay == 'All') {
-                     $scope.lastIndex = 1;
-                     $scope.nodeStatus = nodeStatusCopy;
-                 } else {
-                     $scope.lastIndex = Math.ceil(nodeStatusCopy.length / $scope.RecordsToDisplay);
-                     $scope.nodeStatus = nodeStatusCopy.slice(0, $scope.RecordsToDisplay);
-                 }
-                 $scope.currentPage = 1;
+                $scope.UpdateRecordsToShow = () => {
+                    if($scope.RecordsToDisplay == 'All') {
+                        $scope.lastIndex = 1;
+                        $scope.nodeStatus = nodeStatusCopy;
+                    } else {
+                        $scope.lastIndex = Math.ceil(nodeStatusCopy.length / $scope.RecordsToDisplay);
+                        $scope.nodeStatus = nodeStatusCopy.slice(0, $scope.RecordsToDisplay);
+                }
+                $scope.currentPage = 1;
             }
             /* Page Slicing  logic */
-            $scope.handlePagination = (pageIndex, isDisabled) => {
-                if(!isDisabled) {
-                    let startIndex = 0, endIndex = 0;
-                    $scope.currentPage = pageIndex;
-                    startIndex = (pageIndex * $scope.RecordsToDisplay) - $scope.RecordsToDisplay;
-                    endIndex = startIndex + parseInt($scope.RecordsToDisplay);
-                    $scope.nodeStatus = nodeStatusCopy.slice(startIndex, endIndex);
+                $scope.handlePagination = (pageIndex, isDisabled) => {
```

Comment:
There is a significant amount of unnecessary indentation added in scm.js. Could you please remove it?